scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/synchronization/AOSynchronizationTest.java

```diff
@@ -0,0 +1,469 @@
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.synchronization;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.log4j.PatternLayout;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.objectweb.proactive.ActiveObjectCreationException;
+import org.objectweb.proactive.api.PAActiveObject;
+import org.objectweb.proactive.core.config.CentralPAPropertyRepository;
+import org.objectweb.proactive.core.node.NodeException;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.job.JobIdImpl;
+import org.ow2.proactive.scheduler.task.TaskIdImpl;
+import org.ow2.tests.ProActiveTestClean;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 05/04/2018
+ */
+public class AOSynchronizationTest extends ProActiveTestClean {
+
+    private static final String CHANNEL1 = "Channel1";
+
+    private static final String CHANNEL2 = "Channel2";
+
+    private static final String USER = "user";
+
+    private static final TaskId TASK_ID = TaskIdImpl.createTaskId(new JobIdImpl(0, "Job"), "Task", 0);
+
+    public static final String PREDICATE_WITH_RUNTIME_ERROR = "{k, x -> 1 / 0}";
+
+    public static final String PREDICATE_WITH_COMPILATION_ERROR = "{k, x -> ?}";
+
+    public static final String BIFUNCTION_WITH_COMPILATION_ERROR = "{k, x -> a +-= 1}";
+
+    public static final String BIFUNCTION_WITH_RUNTIME_ERROR = "{k, x -> 1/0}";
+
+    public static final String FUNCTION_WITH_RUNTIME_ERROR = "{k -> 1/0}";
+
+    public static final String FUNCTION_WITH_COMPILATION_ERROR = "{k -> a +-= 1}";
```

Comment:
Can't you merge `FUNCTION_WITH_COMPILATION_ERROR` and `CONSUMER_WITH_COMPILATION_ERROR`?

Reply:
ok

Reply:
though consumer opers on values where function opers on keys, this is why the parameter name is different

Reply:
Then I think it is clearer to keep them separated.